sstables/component_type.hh

```diff
@@ -68,3 +76,11 @@ struct fmt::formatter<sstables::component_type> : fmt::formatter<string_view> {
         }
     }
 };
+
+template <>
+struct fmt::formatter<sstables::component_name> : fmt::formatter<string_view> {
+    template <typename FormatContext>
+    auto format(const sstables::component_name& name, FormatContext& ctx) const {
+        return fmt::format_to(ctx.out(), "{}", name.format());
+    }
```

Comment:
could simplify this further like:
```c++
auto format(const sstables::component_name& name, auto& ctx) const {
        return fmt::format_to(ctx.out(), "{}", name.format());
    }
}
```
but, i am wondering why don't just let the formatter do its job, so that it looks like:
```c++
auto format(const sstables::component_name& name, auto& ctx) const {
        return fmt::format_to(ctx.out(), "{}/{}", sst._storage->prefix(), sst.component_basename(component));
    }
}
```

so we don't create a temporary instance of `sstring` and then drop it on the floor only to concatenate it with another sstring. if there is any potential users of the `format()` member function, they could use `seastar::to_string()` or `fmt::to_string()`. or better off using the formatter directly .

Reply:
The reason is

```
./sstables/component_type.hh:84:59: error: member access into incomplete type 'const sstable'
   84 |         return fmt::format_to(ctx.out(), "{}/{}", name.sst._storage->prefix(), name.sst.component_basename(name.component));
      |                                                           ^
```

We'd need to include a heavyweight sstables.hh into lightweight component_type.hh